src/ape/utils/os.py

```diff
@@ -369,3 +370,69 @@ def extract_archive(archive_file: Path, destination: Optional[Path] = None):
 
     else:
         raise ValueError(f"Unsupported zip format: '{archive_file.suffix}'.")
+
+
+class CacheDirectory:
+    """
+    A directory for caching data where each data item is named
+    ``<key>.json`` and is in the directory. You can access the
```

Comment:
should this have some prefix like `cache_<key>.json`?

Reply:
maybe, or all in `/caches` sub-folder but i dont want to change the structure now as it will be too breaking